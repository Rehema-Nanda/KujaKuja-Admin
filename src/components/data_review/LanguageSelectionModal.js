import React from "react";
import {
    Button, Modal, ModalHeader, ModalBody, Row, Col,
} from "reactstrap";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import PropTypes from "prop-types";

import LanguageOptions from "./LanguageOptions";

const LanguageSelectionModal = ({
    isOpen, toggle, selectedForUpdate, handleLanguageSelect, responsesToUpdate,
}) => {
    let language;
    if (responsesToUpdate.length > 0) {
        const langObject = LanguageOptions.find((lng) => lng.value === responsesToUpdate[0].ideaLanguage);
        language = langObject.name;
    }

    return (
        <Modal isOpen={isOpen} toggle={toggle} size="lg">
            <ModalHeader toggle={toggle}>
                Please select a new Language
            </ModalHeader>
            <ModalBody>
                <Row>
                    <Col xs="6">
                        From:
                        <div className="sp-selection-modal-container-from" style={{ height: "200px" }}>
                            <p>{language}</p>
                        </div>
                    </Col>
                    <Col xs="6">
                        <div>
                            To:
                            <div className="lng-selection-modal-container-to">
                                <div className="lng-options">
                                    {LanguageOptions.map((lng) => {
                                        return (
                                            <Button
                                                onClick={() => handleLanguageSelect(lng)}
                                                className="sp-selection-modal-button"
                                                key={lng.value}
                                                value={lng.value}
                                                active={lng.value === selectedForUpdate.ideaLanguage}
                                            >
                                                <span style={{ marginRight: "5%" }}>{lng.name}</span>
                                                <div className="sp-option-caret">
                                                    <FontAwesomeIcon icon="chevron-right" />
                                                </div>
                                            </Button>
                                        );
                                    })}
                                </div>
                            </div>
                        </div>
                    </Col>
                </Row>
            </ModalBody>
        </Modal>
    );
};

LanguageSelectionModal.propTypes = {
    isOpen: PropTypes.bool,
    handleLanguageSelect: PropTypes.func.isRequired,
    responsesToUpdate: PropTypes.arrayOf(PropTypes.object),
    toggle: PropTypes.func.isRequired,
    selectedForUpdate: PropTypes.objectOf(PropTypes.any),
};

LanguageSelectionModal.defaultProps = {
    isOpen: false,
    responsesToUpdate: [],
    selectedForUpdate: {},
};

export default LanguageSelectionModal;
