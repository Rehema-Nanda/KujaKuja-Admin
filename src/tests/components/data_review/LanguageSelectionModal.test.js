import React from "react";
import { mount } from "enzyme";

import LanguageSelectionModal from "../../../components/data_review/LanguageSelectionModal";

describe("LanguageSelectionModal component", () => {
    const testProps = {
        isOpen: true,
        toggle: jest.fn(),
        selectedForUpdate: {
            columnName: "",
            name: "",
            value: {},
        },
        responsesToUpdate: [{ id: "1", ideaLanguage: "en" }],
        handleLanguageSelect: jest.fn(),
    };

    const wrapper = mount(<LanguageSelectionModal {...testProps} />);

    it("should render correctly", () => {
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it("should call handler function on language select", () => {
        const toggleBtn = wrapper.find("button").at(1);
        expect(toggleBtn.text()).toBe("English");

        toggleBtn.simulate("click");
        expect(testProps.handleLanguageSelect).toHaveBeenCalled();
    });
});
