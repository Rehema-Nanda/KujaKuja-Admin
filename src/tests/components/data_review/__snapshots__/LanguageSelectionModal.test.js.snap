// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSelectionModal component should render correctly 1`] = `
"<LanguageSelectionModal isOpen={true} toggle={[Function: mockConstructor]} selectedForUpdate={{...}} responsesToUpdate={{...}} handleLanguageSelect={[Function: mockConstructor]}>
  <Modal isOpen={true} toggle={[Function: mockConstructor]} size=\\"lg\\" autoFocus={true} centered={false} role=\\"dialog\\" backdrop={true} keyboard={true} zIndex={1050} fade={true} onOpened={[Function: noop]} onClosed={[Function: noop]} modalTransition={{...}} backdropTransition={{...}}>
    <Portal node={{...}}>
      <Portal containerInfo={{...}}>
        <div className=\\"\\">
          <Fade onClick={[Function: bound handleBackdropClick]} onMouseDown={[Function: bound handleBackdropMouseDown]} onKeyUp={[Function: bound handleEscape]} onKeyDown={[Function: bound handleTab]} style={{...}} aria-labelledby={[undefined]} role=\\"dialog\\" tabIndex=\\"-1\\" in={true} mountOnEnter={false} unmountOnExit={false} appear={true} enter={true} exit={true} onEnter={[Function: noop]} onEntering={[Function: noop]} onEntered={[Function: bound onOpened]} onExit={[Function: noop]} onExiting={[Function: noop]} onExited={[Function: bound onClosed]} tag=\\"div\\" baseClass=\\"fade\\" baseClassActive=\\"show\\" timeout={300} cssModule={[undefined]} className=\\"modal\\" innerRef={[undefined]}>
            <Transition onExited={[Function: bound onClosed]} onExiting={[Function: noop]} onExit={[Function: noop]} onEntered={[Function: bound onOpened]} onEntering={[Function: noop]} onEnter={[Function: noop]} timeout={300} exit={true} enter={true} appear={true} unmountOnExit={false} mountOnEnter={false} in={true}>
              <div className=\\"modal fade\\" onClick={[Function: bound handleBackdropClick]} onMouseDown={[Function: bound handleBackdropMouseDown]} onKeyUp={[Function: bound handleEscape]} onKeyDown={[Function: bound handleTab]} style={{...}} aria-labelledby={[undefined]} role=\\"dialog\\" tabIndex=\\"-1\\">
                <div className=\\"modal-dialog modal-lg\\" role=\\"document\\">
                  <div className=\\"modal-content\\">
                    <ModalHeader toggle={[Function: mockConstructor]} tag=\\"h5\\" wrapTag=\\"div\\" closeAriaLabel=\\"Close\\" charCode={215}>
                      <div className=\\"modal-header\\">
                        <h5 className=\\"modal-title\\">
                          Please select a new Language
                        </h5>
                        <button type=\\"button\\" onClick={[Function: mockConstructor]} className=\\"close\\" aria-label=\\"Close\\">
                          <span aria-hidden=\\"true\\">
                            ×
                          </span>
                        </button>
                      </div>
                    </ModalHeader>
                    <ModalBody tag=\\"div\\">
                      <div className=\\"modal-body\\">
                        <Row tag=\\"div\\">
                          <div className=\\"row\\">
                            <Col xs=\\"6\\" tag=\\"div\\" widths={{...}}>
                              <div className=\\"col-6\\">
                                From:
                                <div className=\\"sp-selection-modal-container-from\\" style={{...}}>
                                  <p>
                                    English
                                  </p>
                                </div>
                              </div>
                            </Col>
                            <Col xs=\\"6\\" tag=\\"div\\" widths={{...}}>
                              <div className=\\"col-6\\">
                                <div>
                                  To:
                                  <div className=\\"lng-selection-modal-container-to\\">
                                    <div className=\\"lng-options\\">
                                      <Button onClick={[Function: onClick]} className=\\"sp-selection-modal-button\\" value=\\"en\\" active={false} color=\\"secondary\\" tag=\\"button\\">
                                        <button type=\\"button\\" onClick={[Function: bound onClick]} value=\\"en\\" className=\\"sp-selection-modal-button btn btn-secondary\\" aria-label={{...}}>
                                          <span style={{...}}>
                                            English
                                          </span>
                                          <div className=\\"sp-option-caret\\">
                                            <FontAwesomeIcon icon=\\"chevron-right\\">
                                              <i className=\\"fa\\" />
                                            </FontAwesomeIcon>
                                          </div>
                                        </button>
                                      </Button>
                                      <Button onClick={[Function: onClick]} className=\\"sp-selection-modal-button\\" value=\\"es\\" active={false} color=\\"secondary\\" tag=\\"button\\">
                                        <button type=\\"button\\" onClick={[Function: bound onClick]} value=\\"es\\" className=\\"sp-selection-modal-button btn btn-secondary\\" aria-label={{...}}>
                                          <span style={{...}}>
                                            Español
                                          </span>
                                          <div className=\\"sp-option-caret\\">
                                            <FontAwesomeIcon icon=\\"chevron-right\\">
                                              <i className=\\"fa\\" />
                                            </FontAwesomeIcon>
                                          </div>
                                        </button>
                                      </Button>
                                      <Button onClick={[Function: onClick]} className=\\"sp-selection-modal-button\\" value=\\"sw\\" active={false} color=\\"secondary\\" tag=\\"button\\">
                                        <button type=\\"button\\" onClick={[Function: bound onClick]} value=\\"sw\\" className=\\"sp-selection-modal-button btn btn-secondary\\" aria-label={{...}}>
                                          <span style={{...}}>
                                            Swahili
                                          </span>
                                          <div className=\\"sp-option-caret\\">
                                            <FontAwesomeIcon icon=\\"chevron-right\\">
                                              <i className=\\"fa\\" />
                                            </FontAwesomeIcon>
                                          </div>
                                        </button>
                                      </Button>
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </Col>
                          </div>
                        </Row>
                      </div>
                    </ModalBody>
                  </div>
                </div>
              </div>
            </Transition>
          </Fade>
          <Fade in={true} mountOnEnter={true} unmountOnExit={false} appear={true} enter={true} exit={true} onEnter={[Function: noop]} onEntering={[Function: noop]} onEntered={[Function: noop]} onExit={[Function: noop]} onExiting={[Function: noop]} onExited={[Function: noop]} tag=\\"div\\" baseClass=\\"fade\\" baseClassActive=\\"show\\" timeout={150} cssModule={[undefined]} className=\\"modal-backdrop\\">
            <Transition onExited={[Function: noop]} onExiting={[Function: noop]} onExit={[Function: noop]} onEntered={[Function: noop]} onEntering={[Function: noop]} onEnter={[Function: noop]} timeout={150} exit={true} enter={true} appear={true} unmountOnExit={false} mountOnEnter={true} in={true}>
              <div className=\\"modal-backdrop fade\\" />
            </Transition>
          </Fade>
        </div>
      </Portal>
    </Portal>
  </Modal>
</LanguageSelectionModal>"
`;
